src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -863,7 +1049,18 @@ private static unsafe nuint GetIndexOfFirstNonAsciiChar_Sse2(char* pBuffer, nuin
         FoundNonAsciiDataInFirstVector:
 
             // See comment earlier in the method for an explanation of how the below logic works.
-            currentMask = (uint)Sse2.MoveMask(Sse2.AddSaturate(firstVector, asciiMaskForAddSaturate).AsByte());
+            if (Sse2.IsSupported)
+            {
+                currentMask = (ulong)Sse2.MoveMask(Sse2.AddSaturate(firstVector, asciiMaskForAddSaturate).AsByte());
+            }
+            else if (AdvSimd.Arm64.IsSupported)
+            {
+                currentMask = GetMaskOfMostSignificantBits(AdvSimd.AddSaturate(firstVector, asciiMaskForAddSaturate).AsByte());
+            }
+            else
+            {
+                throw new PlatformNotSupportedException();
+            }
 
         FoundNonAsciiDataInCurrentMask:
 
```

Comment:
3 lines below this at L1069, we're doing `currentMask &= NonAsciiDataSeenMask`. I think this will be messed up now because I made `currentMask` long? I don't know how to fix this yet though. @kunalspathak